RecDP/pyrecdp/primitives/llmutils/text_fixer.py

```diff
@@ -0,0 +1,275 @@
+        # \def\macro_name{macro_value}
+        # where macro_name is only allowed to contain letters and numbers;
+        # macro_value can contain any character.
+        pattern=r'\\def\s*(\\[a-zA-Z0-9]+?)\s*\{(.*?)\}$',
+        flags=re.MULTILINE
+    )
+
+    # Extract all user-defined LaTeX macros from the preamble
+    macros = {}
+    for reg in [non_arg_nc_reg, non_arg_def_reg]:
+        for match in reg.finditer(file_content):
+            # convert the macro name and value to a raw string that can be
+            # used in re.sub
+            macro_name = match \
+                .group(1).encode("unicode-escape").decode("utf-8")
+            macro_val = match \
+                .group(2).encode("unicode-escape").decode("utf-8")
+
+            macros[macro_name] = macro_val
+
+    return macros
+
+
+def _clean_copyright_comments(text: str):
+    r = PAT.search(text)
+    if r:
+        # found one, now see if it contains "copyright", if so strip it
+        span = r.span()
+        sub = text[span[0]:span[1]]
+        if CPAT.search(sub):
+            # cut it
+            text = text[: span[0]] + text[span[1]:]
+
+        return text
+
+    lines = text.split('\n')
+    skip = 0
+
+    # Greedy replace any file that begins with comment block, most
+    # are copyright headers
+    for k in range(len(lines)):
+        if (
+                lines[k].startswith("//") or
+                lines[k].startswith("#") or
+                lines[k].startswith("--") or
+                not lines[k]
+        ):
+            skip = skip + 1
+        else:
+            break
+
+    if skip:
+        # we skipped, consume it
+        text = "\n".join(lines[skip:])
+    return text
+
+
+def get_fixer_by_type(text_type):
+    if isinstance(text_type, str):
+        text_type = getattr(SupportedType, text_type.upper())
+    if SupportedType.HTML == text_type:
+        return clean_html
+    if SupportedType.LATEX == text_type:
+        return clean_latex
+    if SupportedType.CODES == text_type:
+        return clean_codes
+
+
+def text_fixer(data_dir, in_type, out_dir, text_types, enable_ray=False):
+    if enable_ray:
+        rdp = SparkDataProcessor(spark_mode='ray')
+    else:
+        rdp = SparkDataProcessor()
+    spark = rdp.spark
+    try:
+        with Timer(f"Load data from {in_type} file"):
+
+            data_files = get_target_file_list(data_dir, in_type)
+            data_files = [os.path.join(data_dir, f) for f in data_files]
+            if in_type == 'parquet':
+                spark_df = read_parquet(data_files, spark)
+            elif in_type == 'jsonl':
+                spark_df = read_json(data_files, spark)
+            total_data_num = spark_df.count()
+
+        with Timer("Processing data"):
+            fixed_df = spark_df
+            for text_type in text_types:
+                operator = get_fixer_by_type(text_type)
+                fixed_df = fixed_df.withColumn('text', operator(F.col('text')))
```

Comment:
make this as a new function, so we can also support lazy execution to link every steps together
text_fixer_spk(spark_df): -> spark_df